### PR TITLE
Refactored Project Structure and Files

### PR DESCRIPTION
**Pull request by DudeBro249**
Added environment variable support, docstrings for most functions, and changed folder structure for organization.
Fixed Issues 1 and 8

### CODE REVIEW

File: src/api/.env.example
Comment (GDWR) [inline src/api/.env.example:1]: I would prefer for this example to be in the README.

File: src/api/requirements.txt
Comment (GDWR) [inline src/api/requirements.txt:9]: I have removed the usage of orm,
Can you remove this and other relevant packages it uses.

File: src/bot/.env.example
Comment (GDWR) [inline src/bot/.env.example:2]: I would prefer for this to be in a README

File: src/bot/.gitignore
Comment (GDWR) [inline src/bot/.gitignore:1]: How does multiple
.gitignore files work?
Do they apply to the subdirectories?

File: src/bot/utils/load_env.py
Comment (GDWR) [inline src/bot/utils/load_env.py:4]: Shared utilities would be a good thing rather than duplicating code.

This might be impossible to do cleanly but it would be nice.

Can you have a think of how this could be done and let me know?

**Review by GDWR — COMMENTED** [review]
I have look through and it all looks good,
Left some comments if like so responses too and
Additionally I'd like to get this up in my IDE before merge.

Good stuff tho!

File: src/api/requirements.txt
Comment (GDWR) [reply src/api/requirements.txt:9]: This isn't massively important

Comment (DudeBro249) [reply src/api/requirements.txt:9]: Sure, I'll do that

File: src/bot/.env.example
Comment (DudeBro249) [reply src/bot/.env.example:2]: I will put environment variable instructions in the readme _**as well**_ , but I think it would also be nice to have a .env.example file(or some other name) as well within the repository because that is a standard in the development community.

File: src/bot/.gitignore
Comment (DudeBro249) [reply src/bot/.gitignore:1]: They do

File: src/bot/utils/load_env.py
Comment (DudeBro249) [reply src/bot/utils/load_env.py:4]: It actually depends on the deployment strategy of the api and the bot.

File: src/bot/.env.example
Comment (GDWR) [reply src/bot/.env.example:2]: if that is a standard then we can keep it, I am not used to contributing projects. 
Thanks for letting me know.

File: src/bot/utils/load_env.py
Comment (GDWR) [reply src/bot/utils/load_env.py:4]: I will discuss this with you via voice

File: src/bot/.env.example
Comment (DudeBro249) [reply src/bot/.env.example:2]: I've made the necessary changes to the README.md file. Please check it out whenever you get a chance.

File: src/api/requirements.txt
Comment (DudeBro249) [reply src/api/requirements.txt]: I've done the needful.

File: src/api/.env.example
Comment (DudeBro249) [reply src/api/.env.example:1]: I've done the needful. Please check the README file whenever you get a chance.

**Review by GDWR — APPROVED** [review]

**Comment by GDWR** [comment]
Please resolve the conflict, sorry!
I will merge once they have been resolved.

**Comment by DudeBro249** [comment]
I changed the usage of the `setting` function slightly from what was specified on [this issue](https://github.com/GDWR/DiscordArena/issues/1).
The usage is now:
```python
TOKEN = setting("TOKEN")
```
There is no `required` keyword argument and `default` has become an optional keyword argument. If default is not passed, the function raises an `EnvironmentError`. This is to ensure that we do not get any `None` values for our environment constants.
Let me know what you think.

**Review by GDWR — APPROVED** [review]

**Review by GDWR — APPROVED** [review]

**Comment by GDWR** [comment]
@DudeBro249 In the future please do not add code that is already being worked on.
if you wish to test something from another issue, create a new branch.

The reason is right now I want to merge your branch, but I want to discuss with Jake what he thinks about this approach you have done for settings and how we could and a required parameter.

For this instance it is fine, but in the future please be aware of this.

**Comment by DudeBro249** [comment]
Got it, thanks

**Pull request merged**
